transitions/index.js

```diff
@@ -242,18 +246,62 @@ var applyTransitions = function(options) {
             change: options.change,
             audit: options.audit,
             results: results
+        }, callback);
+    });
+};
+
+var getMetaData = function(callback) {
+    db.medic.get('sentinel-meta-data', function(err, doc) {
+        if (err) {
+            if (err.statusCode !== 404) {
+                return callback(err);
+            }
+            doc = {
+                _id: 'sentinel-meta-data',
+                processed_seq: 0
+            };
+        }
+        callback(null, doc);
+    });
+};
+
+var getProcessedSeq = function(callback) {
+    getMetaData(function(err, doc) {
+        if (err) {
+            logger.error('Error getting meta data', err);
+            return callback(err);
+        }
+        callback(null, doc.processed_seq);
+    });
+};
+
+var updateMetaData = function(seq, callback) {
+    getMetaData(function(err, metaData) {
+        if (err) {
+            logger.error('Error updating metaData', err);
+            return callback();
+        }
+        metaData.processed_seq = seq;
+        db.medic.insert(metaData, function(err) {
+            if (err) {
+                logger.error('Error updating metaData', err);
+            }
+            return callback();
         });
     });
 };
 
 var attach = function() {
 
-    var db = require('../db'),
-        audit = require('couchdb-audit')
+    var audit = require('couchdb-audit')
             .withNano(db, db.settings.db, db.settings.auditDb, db.settings.ddoc, db.settings.username);
 
-    var backlog_delay = 500; // ms
-    var progress_interval = 500; // items
+    var processed = 0;
+    var changeQueue = async.queue(function(task, callback) {
+        processChange(task, function() {
+            _.delay(callback, PROCESSING_DELAY);
+        });
+    });
 
     var match_types = [
```

Comment:
I know you didn't change anything here, but this is better as a constant at the top.